src/plugins/intel_gna/src/transformations/insert_identity_layer.cpp

```diff
@@ -231,3 +232,98 @@ bool IdentityCandidatesCleanup::run_on_model(const std::shared_ptr<ov::Model>& f
     }
     return false;
 }
+
+bool InsertIdentityForPrecAgnosticConcatInput::all_inputs_point_the_same_node(const shared_ptr<ov::Node>& node) {
+    const auto& inputs = node->inputs();
+    const auto& input0 = inputs[0].get_tensor_ptr();
+    return all_of(inputs.begin(), inputs.end(), [&](const ov::Input<ov::Node>& in) {
+        return in.get_tensor_ptr() == input0;
+    });
+}
+
+size_t InsertIdentityForPrecAgnosticConcatInput::find_prev_layer_output_index(shared_ptr<ov::Node> prev,
+                                                                              shared_ptr<ov::Node> next) {
+    for (const auto& output : prev->outputs()) {
+        for (const auto& input : output.get_target_inputs()) {
+            if (input.get_node() == next.get()) {
+                return output.get_index();
+            }
+        }
+    }
+    return 0;
+}
+
+bool InsertIdentityForPrecAgnosticConcatInput::insert_identity_after_nodes(const vector<shared_ptr<ov::Node>>& nodes,
+                                                                           const shared_ptr<ov::Node>& next) {
+    bool is_graph_modified = false;
+    for (auto node : nodes) {
+        size_t index = find_prev_layer_output_index(node, next);
+        insert_identity_layer_after(node, index);
+        is_graph_modified = true;
+    }
+    return is_graph_modified;
```

Comment:
this value will be always true if there will be at least one node on nodes list.

Reply:
Changed to `return nodes.size() > 0`